### ISSUE TITLE
feat: add dynamic route titles in `FAST-Router`

### ISSUE THREAD

**Issue by KingOfTac** [microsoft/fast#6325]
# 🙋 Feature Request

A nice feature I think would be the ability to interpolate route parameters into the title.

## 🤔 Expected Behavior

The ability to do something like this:
```ts
{ path: 'products/{productName}', title: '{productName}' }
```

## 😯 Current Behavior

The above code would produce "{productName}" as the page title.

## 💁 Possible Solution
Since this is all happening in the route mapping, I think all that is required is to check if the `childTitle` contains a match for the route parameter and interpolate the value. Much like it already does for the path.

## 🔦 Context

A common pattern for things like products and blog posts is to set the page title to the title of the product or post in addition to some other text.

## 💻 Examples

Newegg product listing
![image](https://user-images.githubusercontent.com/32497422/186972641-de5cc42f-e838-4d45-9af9-66205ccc9f1e.png)

Post on Medium
![image](https://user-images.githubusercontent.com/32497422/186972820-f0fd3336-d66b-4654-b4bf-db77a7fdf291.png)

**Comment by KingOfTac** [comment]
I self-assigned this; however, I'd like to get other people's thoughts on the feature before starting on anything.

**Comment by cedx** [comment]
Yes, we definitely need it. I'm currently using a dedicated component just to update the title... not very optimized (as a result, the document title is updated twice).

**Comment by KingOfTac** [comment]
> Yes, we definitely need it. I'm currently using a dedicated component just to update the title... not very optimized (as a result, the document title is updated twice).

I created a method to do this recently in another project that involves hooking into the router's internals and creating a custom title builder. Unfortunately, it's hard to do right now because the types needed are all internal, **but** the router could be updated to export the interfaces and types.

I can submit a PR for review by the team sometime this week.
